File: 2284/main.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution
{
public:
    string largestWordCount(vector<string> &messages, vector<string> &senders)
    {
        unordered_map<string, int> m;
        int n = messages.size();
        for (int i = 0; i < n; i++)
        {
            m[senders[i]] += count(messages[i].begin(), messages[i].end(), ' ') + 1;
        }
        int maxWord = 0;
        string ans;
        for (auto &[k, v] : m)
        {
            if (v > maxWord)
            {
                ans = k;
                maxWord = v;
            }
            else if (v == maxWord)
            {
                if (k > ans)
                {
                    ans = k;
                }
            }
        }
        return ans;
    }
};

void solve()
{
    Solution *s = new Solution();
}
int main()
{
    solve();
    return 0;
}

File: 2284/main.swift
class Solution {
    func largestWordCount(_ messages: [String], _ senders: [String]) -> String {
        var nameDict: [String:Int] = [:]
        for (i, sender) in senders.enumerated() {
            var msgCount = messages[i].count(" ") + 1
            nameDict[sender] = (nameDict[sender] ?? 0) + msgCount
        }
        var maxCount = 0
        var ans = ""
        for (k, v) in nameDict {
            if (v > maxCount) {
                maxCount = v
                ans = k
            } else if (v == maxCount) {
                if (k > ans) {
                    ans = k
                }
            }
        }
        return ans
    }
}

extension String {

    // 统计字符出现次数
    func count(_ ch: Character) -> Int {
        var count = 0
        for c in self {
            if c == ch {
                count += 1
            }
        }
        return count
    }
}